sympy/solvers/solveset.py

```diff
@@ -714,6 +716,18 @@ def _solveset(f, symbol, domain, _check=False):
         elif rhs_s is not S.EmptySet:
             result = ConditionSet(symbol, Eq(f, 0), domain)
 
+    if isinstance(result, ConditionSet):
+        num, den = f.as_numer_denom()
+        if not isinstance(den, Number):
+            _result = _solveset(num, symbol, domain)
+            singularities = _solveset(den, symbol, domain)
+            to_check = Intersection(result, singularities)
+            if isinstance(to_check, EmptySet):
+                result = _result
+            elif isinstance(to_check, FiniteSet):
+                to_remove = FiniteSet([s for s in to_check if not (limit(f,symbol,s) == 0)])
```

Comment:
I solve the numerator = 0 and then exclude the solutions that aren't valid because the denominator is zero there too. This doesn't need to be done if it's something like `(x-1)**2/((x-1)*(x-2))` - here `x=1` is actually a solution after cancelling out a factor of `(x-1)` and that's what I check with the limit. On second thoughts, maybe this kind of thing would be better done with `continuous_domain(1/den, symbol, domain)` and intersecting the result with the solutions of the numerator?